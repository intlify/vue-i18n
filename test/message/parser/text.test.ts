import {
  createParser,
  NodeTypes,
  MessageNode,
  TextNode
} from '../../../src/message/parser'

let spy
beforeEach(() => {
  spy = jest.fn()
})

describe('simple', () => {
  test('hello,world', () => {
    const text = 'hello,world'
    const parser = createParser({ onError: spy })
    const ast = parser.parse(text)

    expect(ast).toMatchSnapshot()
    expect(spy).not.toHaveBeenCalled()

    expect(ast.type).toEqual(NodeTypes.Resource)
    expect(ast.body.type).toEqual(NodeTypes.Message)
    const message = ast.body as MessageNode
    expect(message.items).toHaveLength(1)
    const item = message.items[0] as TextNode
    expect(item).toMatchObject({
      type: NodeTypes.Text,
      value: text
    })
  })
})

describe('multi bytes', () => {
  test('こんにちは、世界', () => {
    const text = 'こんにちは、世界'
    const parser = createParser({ onError: spy })
    const ast = parser.parse(text)

    expect(ast).toMatchSnapshot()
    expect(spy).not.toHaveBeenCalled()
    expect(ast.type).toEqual(NodeTypes.Resource)
    expect(ast.body.type).toEqual(NodeTypes.Message)
    const message = ast.body as MessageNode
    expect(message.items).toHaveLength(1)
    const item = message.items[0] as TextNode
    expect(item).toMatchObject({
      type: NodeTypes.Text,
      value: text
    })
  })
})

describe('emoji', () => {
  test('😺', () => {
    const text = '😺'
    const parser = createParser({ onError: spy })
    const ast = parser.parse(text)

    expect(ast).toMatchSnapshot()
    expect(spy).not.toHaveBeenCalled()
    expect(ast.type).toEqual(NodeTypes.Resource)
    expect(ast.body.type).toEqual(NodeTypes.Message)
    const message = ast.body as MessageNode
    expect(message.items).toHaveLength(1)
    const item = message.items[0] as TextNode
    expect(item).toMatchObject({
      type: NodeTypes.Text,
      value: text
    })
  })
})

describe('included end paren', () => {
  test('hello :-)', () => {
    const text = 'hello :-)'
    const parser = createParser({ onError: spy })
    const ast = parser.parse(text)

    expect(ast).toMatchSnapshot()
    expect(spy).not.toHaveBeenCalled()
    expect(ast.type).toEqual(NodeTypes.Resource)
    expect(ast.body.type).toEqual(NodeTypes.Message)
    const message = ast.body as MessageNode
    expect(message.items).toHaveLength(1)
    const item = message.items[0] as TextNode
    expect(item).toMatchObject({
      type: NodeTypes.Text,
      value: text
    })
  })
})

describe('empty string', () => {
  test(`''`, () => {
    const text = ''
    const parser = createParser({ onError: spy })
    const ast = parser.parse(text)

    expect(ast).toMatchSnapshot()
    expect(spy).not.toHaveBeenCalled()
    expect(ast.type).toEqual(NodeTypes.Resource)
    expect(ast.body.type).toEqual(NodeTypes.Message)
    const message = ast.body as MessageNode
    expect(message.items).toHaveLength(0)
  })
})

describe('include new line', () => {
  test('hello\nworld', () => {
    const text = 'hello\nworld'
    const parser = createParser({ onError: spy })
    const ast = parser.parse(text)

    expect(ast).toMatchSnapshot()
    expect(spy).not.toHaveBeenCalled()

    expect(ast.type).toEqual(NodeTypes.Resource)
    expect(ast.body.type).toEqual(NodeTypes.Message)
    const message = ast.body as MessageNode
    expect(message.items).toHaveLength(1)
    const item = message.items[0] as TextNode
    expect(item).toMatchObject({
      type: NodeTypes.Text,
      value: text
    })
  })
})

test.todo(`...`)
test.todo(`hypen-nate`)
test.todo(`hello:`)
test.todo(`1 + 1`)
test.todo(`name = foo`)
test.todo(`'single-quote'`)
test.todo(`"double-qoute"`)
test.todo(` hello world `)
test.todo(`hello\\nworld`)
