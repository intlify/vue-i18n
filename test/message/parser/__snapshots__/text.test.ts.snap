// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emoji üò∫ 1`] = `
Object {
  "body": Object {
    "end": 2,
    "items": Array [
      Object {
        "end": 2,
        "loc": Object {
          "end": Object {
            "column": 3,
            "line": 1,
            "offset": 2,
          },
          "start": Object {
            "column": 1,
            "line": 1,
            "offset": 0,
          },
        },
        "start": 0,
        "type": 3,
        "value": "üò∫",
      },
    ],
    "loc": Object {
      "end": Object {
        "column": 3,
        "line": 1,
        "offset": 2,
      },
      "start": Object {
        "column": 1,
        "line": 1,
        "offset": 0,
      },
    },
    "start": 0,
    "type": 2,
  },
  "end": 2,
  "loc": Object {
    "end": Object {
      "column": 3,
      "line": 1,
      "offset": 2,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "start": 0,
  "type": 0,
}
`;

exports[`empty string '' 1`] = `
Object {
  "body": Object {
    "end": 0,
    "items": Array [],
    "loc": Object {
      "end": Object {
        "column": 1,
        "line": 1,
        "offset": 0,
      },
      "start": Object {
        "column": 1,
        "line": 1,
        "offset": 0,
      },
    },
    "start": 0,
    "type": 2,
  },
  "end": 0,
  "loc": Object {
    "end": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "start": 0,
  "type": 0,
}
`;

exports[`include new line hello
world 1`] = `
Object {
  "body": Object {
    "end": 11,
    "items": Array [
      Object {
        "end": 11,
        "loc": Object {
          "end": Object {
            "column": 6,
            "line": 2,
            "offset": 11,
          },
          "start": Object {
            "column": 1,
            "line": 1,
            "offset": 0,
          },
        },
        "start": 0,
        "type": 3,
        "value": "hello
world",
      },
    ],
    "loc": Object {
      "end": Object {
        "column": 6,
        "line": 2,
        "offset": 11,
      },
      "start": Object {
        "column": 1,
        "line": 1,
        "offset": 0,
      },
    },
    "start": 0,
    "type": 2,
  },
  "end": 11,
  "loc": Object {
    "end": Object {
      "column": 6,
      "line": 2,
      "offset": 11,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "start": 0,
  "type": 0,
}
`;

exports[`included end paren hello :-) 1`] = `
Object {
  "body": Object {
    "end": 9,
    "items": Array [
      Object {
        "end": 9,
        "loc": Object {
          "end": Object {
            "column": 10,
            "line": 1,
            "offset": 9,
          },
          "start": Object {
            "column": 1,
            "line": 1,
            "offset": 0,
          },
        },
        "start": 0,
        "type": 3,
        "value": "hello :-)",
      },
    ],
    "loc": Object {
      "end": Object {
        "column": 10,
        "line": 1,
        "offset": 9,
      },
      "start": Object {
        "column": 1,
        "line": 1,
        "offset": 0,
      },
    },
    "start": 0,
    "type": 2,
  },
  "end": 9,
  "loc": Object {
    "end": Object {
      "column": 10,
      "line": 1,
      "offset": 9,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "start": 0,
  "type": 0,
}
`;

exports[`multi bytes „Åì„Çì„Å´„Å°„ÅØ„ÄÅ‰∏ñÁïå 1`] = `
Object {
  "body": Object {
    "end": 8,
    "items": Array [
      Object {
        "end": 8,
        "loc": Object {
          "end": Object {
            "column": 9,
            "line": 1,
            "offset": 8,
          },
          "start": Object {
            "column": 1,
            "line": 1,
            "offset": 0,
          },
        },
        "start": 0,
        "type": 3,
        "value": "„Åì„Çì„Å´„Å°„ÅØ„ÄÅ‰∏ñÁïå",
      },
    ],
    "loc": Object {
      "end": Object {
        "column": 9,
        "line": 1,
        "offset": 8,
      },
      "start": Object {
        "column": 1,
        "line": 1,
        "offset": 0,
      },
    },
    "start": 0,
    "type": 2,
  },
  "end": 8,
  "loc": Object {
    "end": Object {
      "column": 9,
      "line": 1,
      "offset": 8,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "start": 0,
  "type": 0,
}
`;

exports[`simple hello,world 1`] = `
Object {
  "body": Object {
    "end": 11,
    "items": Array [
      Object {
        "end": 11,
        "loc": Object {
          "end": Object {
            "column": 12,
            "line": 1,
            "offset": 11,
          },
          "start": Object {
            "column": 1,
            "line": 1,
            "offset": 0,
          },
        },
        "start": 0,
        "type": 3,
        "value": "hello,world",
      },
    ],
    "loc": Object {
      "end": Object {
        "column": 12,
        "line": 1,
        "offset": 11,
      },
      "start": Object {
        "column": 1,
        "line": 1,
        "offset": 0,
      },
    },
    "start": 0,
    "type": 2,
  },
  "end": 11,
  "loc": Object {
    "end": Object {
      "column": 12,
      "line": 1,
      "offset": 11,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "start": 0,
  "type": 0,
}
`;
